StarFlight/Classes/SCStarFlightPushClient.m

```diff
@@ -38,57 +38,71 @@ - (instancetype)initWithAppID:(NSString *)appID clientSecret:(NSString *)clientS
 
 - (void)registerWithToken:(NSString *)token
 {
-    [self registerWithToken:token clientUUID:nil tags:nil];
+    [self registerWithToken:token clientUUID:nil tags:nil timePreferences:nil];
 }
 
 - (void)registerWithToken:(NSString *)token tags:(NSArray<NSString *> *)tags
 {
-    [self registerWithToken:token clientUUID:nil tags:tags];
+    [self registerWithToken:token clientUUID:nil tags:tags timePreferences:nil];
 }
 
-- (void)registerWithToken:(NSString *)token clientUUID:(NSString *)clientUUID tags:(NSArray<NSString *> *)tags
+- (void)registerWithToken:(NSString *)token
+               clientUUID:(NSString *)clientUUID
+                     tags:(NSArray<NSString *> *)tags
+          timePreferences:(NSDictionary *)timePreferencesDict
 {
     if (self.appID && self.clientSecret && token)
     {
         NSString *post = [NSString stringWithFormat:@"action=register&appId=%@&clientSecret=%@&type=ios&token=%@", self.appID, self.clientSecret, token];
-
-        if (clientUUID)
+        
+        if (clientUUID && ![clientUUID  isEqualToString: @""])
         {
             post = [post stringByAppendingString:[NSString stringWithFormat:@"&clientUuid=%@", clientUUID]];
         }
-
+        
         NSString *tagsToRegister = @"";
         for (NSString *tag in tags)
         {
             tagsToRegister = [tagsToRegister stringByAppendingString:[NSString stringWithFormat:@",%@", tag]];
         }
-
+        
         tagsToRegister = [tagsToRegister stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@", "]];
-
+        
         if (tagsToRegister && ![tagsToRegister isEqualToString:@""])
         {
             post = [post stringByAppendingString:[NSString stringWithFormat:@"&tags=%@", tagsToRegister]];
         }
-
+        
+        if (timePreferencesDict) {
+            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:timePreferencesDict options:NSJSONWritingPrettyPrinted error:nil];
```

Comment:
Check for nil, your code might crash if a nil is found in this method.

Reply:
Please explain more?

Reply:
@ankushkushwaha Your NSData and NSString initWithData can return nil objects if you have a wrong json for example.